src/cmake/thirdparty/SetupAxomThirdParty.cmake

```diff
@@ -196,6 +196,23 @@ else()
     message(STATUS "MFEM support is OFF")
 endif()
 
+# caliper-enabled mfem in cuda configs depends on cupti, which is not properly exported
+if(TARGET mfem AND ENABLE_CUDA)
+    # check if mfem depends on caliper; if so, patch it with cupti
+    get_target_property(_mfem_libs mfem INTERFACE_LINK_LIBRARIES)
+    if("${_mfem_libs}" MATCHES "caliper")
+        if(NOT TARGET CUDA::toolkit)
+            find_package(CUDAToolkit REQUIRED)
+        endif()
+
+        if(TARGET CUDA::toolkit)
+            blt_patch_target(NAME mfem DEPENDS_ON CUDA::toolkit)
+        endif()
+        if(TARGET CUDA::cupti)
+            blt_patch_target(NAME mfem DEPENDS_ON CUDA::cupti)
+        endif()
```

Comment:
For reference, here's what these two targets provide in my configuration:
```
-- [CUDA::toolkit property] 'CUDA::toolkit' is a CMake target
-- [CUDA::toolkit property] INTERFACE_INCLUDE_DIRECTORIES: /usr/tce/packages/cuda/cuda-11.1.1/nvidia/include
-- [CUDA::toolkit property] INTERFACE_LINK_DIRECTORIES: /usr/tce/packages/cuda/cuda-11.1.1/nvidia/lib64
-- [CUDA::toolkit property] INTERFACE_SYSTEM_INCLUDE_DIRECTORIES: /usr/tce/packages/cuda/cuda-11.1.1/nvidia/include
-- [CUDA::toolkit property] NAME: CUDA::toolkit
-- [CUDA::toolkit property] TYPE: INTERFACE_LIBRARY

-- [CUDA::cupti property] 'CUDA::cupti' is a CMake target
-- [CUDA::cupti property] INTERFACE_INCLUDE_DIRECTORIES: /usr/tce/packages/cuda/cuda-11.1.1/nvidia/include
-- [CUDA::cupti property] INTERFACE_LINK_LIBRARIES: /usr/tce/packages/cuda/cuda-11.1.1/nvidia/lib64/libcupti.so
-- [CUDA::cupti property] INTERFACE_SYSTEM_INCLUDE_DIRECTORIES: /usr/tce/packages/cuda/cuda-11.1.1/nvidia/include
-- [CUDA::cupti property] NAME: CUDA::cupti
-- [CUDA::cupti property] TYPE: INTERFACE_LIBRARY
```

Reply:
We should also try to work to get logic like this into into caliper's cmake -- it doesn't handle forwarding deps (CUDA, HIP, gotcha, etc). That's ok for final link for an app that is already handling these deps, but for the case when one lib is linked with bells and whistles, makes things hard.

Reply:
I completely agree and was going to open an issue on caliper

Tag: @daboehme

Reply:
Having said that, I don't think it would help us in this case since we're getting this from mfem's `config.mk` which does not have CMake imports/exports, and axom does not (yet) use caliper.

Reply:
True, however it could also help mfem forward that info via a config.mk of its own :-)